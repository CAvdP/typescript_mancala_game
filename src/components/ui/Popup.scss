#pop-up {
    position: absolute;
    width: 1400px;
    height: 600px;
    z-index: 100;
    left: calc(50% - 700px);
    top: calc(50% - 300px);
    background-color: #100751;
    color: white;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}