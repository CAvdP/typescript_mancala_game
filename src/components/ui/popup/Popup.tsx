import React from 'react';
import './styling/Popup.scss';

interface Props {
player1Points: number;
player2Points: number;
resetFunction: ()=> void;
}

function Pit({player1Points, player2Points, resetFunction}: Props){
    let winner = player1Points > player2Points ? 'Player one': 'Player two'
    let winnerPoints = winner === 'Player one'? player1Points: player2Points

    return (
        <div id='pop-up' className={winner === 'Player one' ? 'neon_blue' :'neon_pink'}>
            <h1>{`${winner} won with ${winnerPoints} points!`} </h1>
            <button className='neon_pink_blue btn' onClick={()=> resetFunction()}>Reset</button>
        </div>
    )
}

export default Pit