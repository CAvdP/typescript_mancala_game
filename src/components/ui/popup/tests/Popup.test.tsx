import React from 'react'
import renderer from 'react-test-renderer'
import Popup from '../Popup';



describe("Popup Component", () => {
    it('matches the saved snapshot', () => {
        const actualSnapshot = renderer
          .create(<Popup player1Points={0} player2Points={0} resetFunction={()=>{}} />)
          .toJSON()
          
        expect(actualSnapshot).toMatchSnapshot()
      });
});