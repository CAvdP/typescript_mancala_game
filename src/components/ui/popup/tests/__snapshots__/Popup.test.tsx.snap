// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Component matches the saved snapshot 1`] = `
<div
  className="neon_pink"
  id="pop-up"
>
  <h1>
    Player two won with 0 points!
     
  </h1>
  <button
    className="neon_pink_blue btn"
    onClick={[Function]}
  >
    Reset
  </button>
</div>
`;
